config/extension_hook.go

```diff
@@ -148,6 +176,16 @@ func (list HookList) Filter(filter HookFilter) HookList {
 	return result
 }
 
+func (config *TerragruntConfig) initializeHooks() {
+	for idx := range config.PreHooks {
+		config.PreHooks[idx].Type = PreHookType
+	}
+
+	for idx := range config.PostHooks {
+		config.PostHooks[idx].Type = PostHookType
+	}
+}
```

Comment:
Would it be possible to integrate that within the two following calls in the convertToTerragruntConfig (161-162) ? :
```golang
tcf.PreHooks.init(tcf)
tcf.PostHooks.init(tcf)
```

I understand it's autogenerated code so it's more a question than a suggestion

Reply:
I think that it would be possible but it's going to require some effort.

The issue with the code you quoted here is that once `init` is called all notion of whether we're in a pre or post hook is lost.

I could change the way `init` works in this context to allow this. The simplest option would be for me to add the hook type as an argument that move on from there. I have a feeling that go is going to fight me on this because there's a bunch of interfaces in place for all this and I'd be changing that.

Another alternative which I have no idea how to implement would be to somehow bake the information on the hook type into the type of the hook list. I could define something like a `PreHookList` and `PostHookList` and then the type system would just resolve the right functions. I've tried implementing something like this but I hit a wall. I don't know if go's type system even allows this. My guess is that it can't.

I like your suggestion tho, I think that I can minimally move the code that initializes the hook types in the same bit of code as where all other `init` functions are called.

Reply:
@wtrep I've done it! 🎉 

The code is now integrated in the initialization block you referenced.